### ISSUE TITLE
Improper clamping behavior on graphic effects

### ISSUE THREAD

**Issue by thisandagain** [LLK/scratch-vm#1283]
### Steps to Reproduce
- https://scratch.mit.edu/projects/228123708/
- https://preview.scratch.mit.edu/#228123708

### Reference via Mr-Dave
```
I looked at your ice cream maker project and I believe that the problem you are having is that Scratch 3.0 does not limit the min and max values for what we can set for the brightness and ghosting settings.  You can see the difference in this small project.  Under Scratch 2.0 the cat and the backdrop recover and look the same no matter that both have had their values set to more that the max or min of 100 and -100.  However, run this same project under Scratch 3.0 and all you'll see is solid black.

Other people have mentioned having the same problem.
```

**Comment by LiFaytheGoblin** [comment]
Related to this is how the effect should behave if it is at the 'edge', like in this case:
<img width="976" alt="screen shot 2018-08-29 at 3 45 16 pm" src="https://user-images.githubusercontent.com/5471273/44811407-a53ba000-aba2-11e8-9d56-d14b122bda72.png">

Project 242576633

**Comment by ktbee** [comment]
closed via #2042
